### PR TITLE
User controller g

### PR DESCRIPTION
**Pull request by edilenedacruz**
This adds and configures shoulda-matchers, adds model validation to user, updates factory for user, adds feature tests for creating an user.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -4,4 +4,6 @@
   get      '/login', to: 'sessions#new', as: "login"
   get      '/signup', to: 'users#new'
   post     '/login', to: 'sessions#create'
+  post     '/signup', to: 'users#create'
```
Comment (CjMoore) [inline config/routes.rb]: I think to be restful we need to change this '/signup' to '/users'? Does the client have an opinion? @s-espinosa @AliSchlereth

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+    it { should have_secure_password }
+  end
+
+  context "validations" do
+    it { should validate_presence_of(:first_name) }
+    it { should validate_presence_of(:last_name) }
+    it { should validate_presence_of(:username) }
+    it { should validate_presence_of(:email) }
+    it { should validate_presence_of(:password) }
+    it { should validate_presence_of(:phone) }
```
Comment (CjMoore) [inline spec/models/user_spec.rb]: Shoulda matchers are the best. ❤️

Comment (edilenedacruz) [inline spec/models/user_spec.rb]: Agreed!!!

File: config/routes.rb (hunk 1 above)
Comment (edilenedacruz) [inline config/routes.rb]: I was thinking the same

File: spec/factories/users.rb [hunk 3]
```diff
@@ -1,9 +1,10 @@
 FactoryGirl.define do
   factory :user do
-    first_name "MyString"
-    last_name "MyString"
-    password_digest "MyString"
-    email "MyString"
-    phone "MyString"
+    first_name { Faker::Name.first_name }
```
Comment (cews7) [inline spec/factories/users.rb]: @edilenedacruz I updated the factory using faker

**Pull request merged**